core-rust/state-manager/src/jni/transaction_store.rs

```diff
@@ -116,24 +116,20 @@ fn do_get_transaction_at_state_version(
     let committed_transaction_receipt = state_manager
         .store()
         .get_committed_transaction_receipt(state_version)?;
-
-    let ledger_receipt_bytes = scrypto_encode(&committed_transaction_receipt).unwrap();
+    let ledger_receipt = committed_transaction_receipt.on_ledger;
+    let local_execution = committed_transaction_receipt.local_execution;
 
     Some(ExecutedTransaction {
-        outcome: match committed_transaction_receipt.outcome {
-            LedgerTransactionOutcome::Success(output) => TransactionOutcomeJava::Success(output),
-            LedgerTransactionOutcome::Failure(err) => {
+        outcome: match local_execution.outcome {
+            DetailedTransactionOutcome::Success(output) => TransactionOutcomeJava::Success(output),
+            DetailedTransactionOutcome::Failure(err) => {
                 TransactionOutcomeJava::Failure(format!("{err:?}"))
             }
         },
-        ledger_receipt_bytes,
+        ledger_receipt_bytes: scrypto_encode(&ledger_receipt.get_consensus_receipt()).unwrap(),
```

Comment:
I think the general direction is good, but there's slightly confusing naming / renaming going on here - we have a:
* CommittedTransactionReceipt
* LocalReceipt
* LedgerReceipt
* ConsensusReceipt
* LocalExecution (which includes both Outcome and ExecutionTrace from the engine receipt)
* In the engine: All the engine receipt names
* In other places in this PR: "full receipt"

I'm not entirely sure I understand clearly what they all are :) - and there are still some consistency clean-ups we need.

I suggest we have a clear "Proposed Naming and Concept Grouping" section of `REP-65` - which covers both the receipts in the Engine, and remapped receipts in the node, which we can all agree on. (and would also cover which things - such as Balance Changes, and Parent information - can be derived from which receipt)

EG I think maybe we just need to call this "consensus_receipt_bytes" in the API and we'd almost be there.

Reply:
Looks great, thanks. I just want to make sure we're all clear on the naming scheme so we can be consistent with it!